launcher/auth/auth.go

```diff
@@ -1,4 +1,5 @@
-package launcher
+// Package auth contains functionalities to authenticate docker repo.
+package auth
```

Comment:
Like having a new Resolver in this package that takes in mds client instead of the token

Reply:
Done.